pilot/pkg/model/config_test.go

```diff
@@ -177,42 +178,82 @@ func TestServiceKey(t *testing.T) {
 func TestSubsetKey(t *testing.T) {
 	hostname := model.Hostname("hostname")
 	cases := []struct {
-		hostname model.Hostname
-		subset   string
-		port     *model.Port
-		want     string
+		gotOverride string
+		hostname    model.Hostname
+		subset      string
+		port        *model.Port
+		want        string
+		err         error
+		inbound     bool
 	}{
 		{
 			hostname: "hostname",
 			subset:   "subset",
 			port:     &model.Port{Name: "http", Port: 80, Protocol: model.ProtocolHTTP},
-			want:     "outbound|http|subset|hostname",
+			want:     "_http._subset.hostname",
```

Comment:
should've raised this before - it seems odd that one illegal character | fails envoy parsing but _ which is also illegal should be ok. are we sure this will be fine?

Reply:
_ is a legal character for dns. It is used in bonjour and it is also legal authority header char per
Nghttp2. https://github.com/nghttp2/nghttp2/blob/842509dab604e6cc62f7fb98d448294c6ec85763/lib/nghttp2_http.c#L331

Reply:
_ is legal for DNS and used in SRV and DNS-SD. It is not legal for hostnames - which is the reason
SRV and DNS-SD are using it in labels for port, protocol, etc